dask/dataframe/core.py

```diff
@@ -1656,9 +1656,11 @@ def sum(
             "sum", axis=axis, skipna=skipna, split_every=split_every, out=out
         )
         if min_count:
-            return result.where(
-                self.notnull().sum(axis=axis) >= min_count, other=np.NaN
-            )
+            cond = self.notnull().sum(axis=axis) >= min_count
+            if is_series_like(cond):
+                return result.where(cond, other=np.NaN)
+            else:
+                return result if cond.compute() else np.NaN
```

Comment:
We'd like to avoid this compute if possible, since it may be expensive.

I think for Series.sum(min_count=1) we'll return a Scalar either way. The Scalars value should be `np.nan` if the count is less than min_count.

Reply:
Makes sense, I was wondering about the compute also. Is there a preferred way to make this "lazy"?